company/uber/SpiralMatrix.java

```diff
@@ -0,0 +1,61 @@
+//Given a matrix of m x n elements (m rows, n columns), return all elements of the matrix in spiral order.
+//
+//Example 1:
+//
+//Input:
+//[
+ //[ 1, 2, 3 ],
+ //[ 4, 5, 6 ],
+ //[ 7, 8, 9 ]
+//]
+//Output: [1,2,3,6,9,8,7,4,5]
+//Example 2:
+//
+//Input:
+//[
+  //[1, 2, 3, 4],
+  //[5, 6, 7, 8],
+  //[9,10,11,12]
+//]
+//Output: [1,2,3,4,8,12,11,10,9,5,6,7]
+
+class SpiralMatrix {
+    public List<Integer> spiralOrder(int[][] matrix) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.